clusterloader2/api/types.go

```diff
@@ -0,0 +1,124 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package api
+
+import (
+	"time"
+)
+
+// Config is a structure that represents configuration
+// for a single test run
+type Config struct {
+	// Namespaces is a number of namespace used by the test.
+	Namespaces int32 `json: namespaces`
+	// Steps is a sequence of test steps.
+	Steps []Step `json: steps`
+	// TuningSets is a collection of tuning sets that can be used by steps.
+	TuningSets []TuningSet `json: tuningSets`
+}
+
+// Step is a structure that represents encapsulation
+// of some actions. This actions could be object declarations
+// or measurement usages.
+type Step struct {
+	// Phases is a collection of declarative definitions of objects
+	Phases []Phase `json: phases`
+	// Measurements is a collection of measurement calls
+	Measurements []string `json: measurements`
+}
+
+// Phase is a structure that declaratively defines state of objects.
+// In given namespace range (all if not specified) it specifies the number
+// and the configuration of managed objects.
+type Phase struct {
+	// NamespaceRange defines the set of namespaces in which objects
+	// should be reconciled.
+	// If null, objects are assumed to be cluster scoped.
+	NamespaceRange *NamespaceRange `json: namespaceRange`
+	// ReplicasPerNamespace is a number of instances of a given object
+	// to exist in each of referenced namespaces.
+	ReplicasPerNamespace int32 `json: replicasPerNamespace`
+	// TuningSet is the name of TuningSet to be used.
+	TuningSet string `json: tuningSet`
+	// Objects is a set of objects that should be reconciled.
+	Objects []Object `json: objects`
+}
+
+// Object is a structure that defines the object managed be the tests
+type Object struct {
+	// ObjectType is a type definition for a given object.
+	ObjectType ObjectType `json: objectType`
+	// Basename is a string from which names of objects will be created.
+	Basename string `json: basename`
+	// ObjectTemplatePath specifies the path to object definition.
+	ObjectTemplatePath string `json: objectTemplatePath`
+}
+
+// NamespaceRange specifies the range of namespaces [Min, Max]
+type NamespaceRange struct {
+	// Min is the lower index of namespace range
+	Min int32 `json: min`
+	// Min is the upper index of namespace range
+	Max int32 `json: max`
```

Comment:
Please take a look into recent changes to design doc.

Reply:
Updated NamespaceRange struct.
Added Measurement struct.